labelbox/orm/model.py

```diff
@@ -239,11 +239,58 @@ class EntityMeta(type):
     of the Entity class object so they can be referenced for example like:
         Entity.Project.
     """
+    mappings: Dict[str, "Entity"] = {}
 
     def __init__(cls, clsname, superclasses, attributedict):
         super().__init__(clsname, superclasses, attributedict)
+        cls.validate_cached_relationships()
         if clsname != "Entity":
             setattr(Entity, clsname, cls)
+            EntityMeta.mappings[utils.snake_case(
+                cls.__name__)] = cls.relationships()
+
+    @staticmethod
+    def raise_for_nested_cache(first: str, middle: str, last: str):
+        raise TypeError(
+            "Cannot cache a relationship to an Entity with its own cached relationship(s). "
+            f"`{first}` caches `{middle}` which caches `{last}`")
+
+    def validate_cached_relationships(cls):
+        """
+        Graphql doesn't allow for infinite nesting in queries. 
+        This function checks that cached relationships result in valid queries.
```

Comment:
Forgive my ignorance (I should have context after reviewing previous PR), but why would caching affect the query in such a way that it would get infinite / invalid?

Reply:
I don't think infinite ToOne relationships work. It was causing problems for me. But this keeps the logic much simpler. If we want to expand this in the future we can.